apps/aeoracle/src/aeo_register_tx.erl

```diff
@@ -0,0 +1,179 @@
+process(#oracle_register_tx{account       = AccountPubKey,
+                            nonce         = Nonce,
+                            fee           = Fee} = RegisterTx, Trees0, Height) ->
+    AccountsTree0 = aec_trees:accounts(Trees0),
+    OraclesTree0  = aec_trees:oracles(Trees0),
+
+    Account0 = aec_accounts_trees:get(AccountPubKey, AccountsTree0),
+    {ok, Account1} = aec_accounts:spend(Account0, Fee, Nonce, Height),
+    AccountsTree1 = aec_accounts_trees:enter(Account1, AccountsTree0),
+
+    Oracle = aeo_oracles:new(RegisterTx, Height),
+    OraclesTree1 = aeo_state_tree:enter_oracle(Oracle, OraclesTree0),
+
+    Trees1 = aec_trees:set_accounts(Trees0, AccountsTree1),
+    Trees2 = aec_trees:set_oracles(Trees1, OraclesTree1),
+
+    {ok, Trees2}.
+
+serialize(#oracle_register_tx{account       = AccountPubKey,
+                              nonce         = Nonce,
+                              query_spec    = QuerySpec,
+                              response_spec = ResponseSpec,
+                              query_fee     = QueryFee,
+                              ttl           = {TTLType, TTLValue},
+                              fee           = Fee}) ->
+    [#{<<"type">> => type()},
+     #{<<"vsn">> => version()},
+     #{<<"account">> => AccountPubKey},
+     #{<<"nonce">> => Nonce},
+     #{<<"query_spec">> => QuerySpec},
+     #{<<"response_spec">> => ResponseSpec},
+     #{<<"query_fee">> => QueryFee},
+     #{<<"ttl">> =>
+           #{<<"type">> => TTLType,
+             <<"value">> => TTLValue}},
+     #{<<"fee">> => Fee}].
+
+deserialize([#{<<"type">>          := ?ORACLE_REGISTER_TX_TYPE},
+             #{<<"vsn">>           := ?ORACLE_REGISTER_TX_VSN},
+             #{<<"account">>       := AccountPubKey},
+             #{<<"nonce">>         := Nonce},
+             #{<<"query_spec">>    := QuerySpec},
+             #{<<"response_spec">> := ResponseSpec},
+             #{<<"query_fee">>     := QueryFee},
+             #{<<"ttl">>           := #{<<"type">>  := TTLType,
+                                        <<"value">> := TTLValue}},
+             #{<<"fee">>           := Fee}]) ->
+    #oracle_register_tx{account       = AccountPubKey,
+                        nonce         = Nonce,
+                        query_spec    = QuerySpec,
+                        response_spec = ResponseSpec,
+                        query_fee     = QueryFee,
+                        ttl           = {TTLType, TTLValue},
+                        fee           = Fee}.
+
+-spec type() -> binary().
+type() ->
+    ?ORACLE_REGISTER_TX_TYPE.
+
+-spec version() -> non_neg_integer().
+version() ->
+    ?ORACLE_REGISTER_TX_VSN.
+
+%% -- Local functions  -------------------------------------------------------
+
+ensure_not_oracle(PubKey, Trees) ->
+    OraclesTree  = aec_trees:oracles(Trees),
+    case aeo_state_tree:lookup_oracle(PubKey, OraclesTree) of
+        {value, _Oracle} -> {error, account_is_already_an_oracle};
```

Comment:
Error reason appear misleading - it may be oracle interaction?

Reply:
No, an oracle interaction will never be returned by lookup_oracle...

Reply:
No. If there is an Id clash there will be an exception from aec_oracles:deserialize/1